src/cdk/v2/destinations/the_trade_desk/rtWorkflow.yaml

```diff
@@ -1,17 +1,17 @@
 bindings:
   - name: processRouterDest
-    path: ./utils
+    path: ./transform
 
 steps:
-  - name: validateInput
+  - name: validateConfig
     template: |
-      $.assert(Array.isArray(^) && ^.length > 0, "Invalid event array")
       const config = ^[0].destination.Config
-      $.assertConfig(config.audienceId, "Segment name is not present. Aborting")
       $.assertConfig(config.advertiserId, "Advertiser ID is not present. Aborting")
       $.assertConfig(config.advertiserSecretKey, "Advertiser Secret Key is not present. Aborting")
+      $.assertConfig(config.trackerId, "Tracking Tag ID is not present. Aborting")
       config.ttlInDays ?  $.assertConfig(config.ttlInDays >=0 && config.ttlInDays <= 180, "TTL is out of range. Allowed values are 0 to 180 days")
 
   - name: processRouterDest
```

Comment:
we could write whole workflow here it self:
```yaml
bindings:
  - name: EventType
    path: ../../../../constants
  - name: processRecordInputs
    path: ./transformRecord
  - name: processConversionInputs
    path: ./transformConversion
  - name: handleRtTfSingleEventError
    path: ../../../../v0/util/index

steps:
  - name: validateConfig
    template: |
      const config = ^[0].destination.Config
      $.assertConfig(config.advertiserId, "Advertiser ID is not present. Aborting")
      $.assertConfig(config.advertiserSecretKey, "Advertiser Secret Key is not present. Aborting")
      $.assertConfig(config.trackerId, "Tracking Tag ID is not present. Aborting")
      config.ttlInDays ?  $.assertConfig(config.ttlInDays >=0 && config.ttlInDays <= 180, "TTL is out of range. Allowed values are 0 to 180 days")

  - name: validateInput
    template: |
      $.assert(Array.isArray(^) && ^.length > 0, "Invalid event array")
  
  - name: processRecordEvents
    template: |
      $.processRecordInputs(^.{.message.type === $.EventType.RECORD})
  
  - name: processConversionEvents
    template: |
      $.processConversionInputs(^.{.message.type === $.EventType.TRACK})
  
  - name: failOtherEvents
    template: |
      ^.{.message.type !== $.EventType.TRACK && .message.type !== RECORD}.map(
        function(event) {
          const error = new InstrumentationError(.message.type +" Event type is not supported");
          $.handleRtTfSingleEventError(event, error, {})
        }
      )

  - name: finalPayload
    template: |
      [...$.outputs.processRecordEvents, ...$.outputs.processConversionEvents, ...$.outputs.failOtherEvents]


```

Reply:
Updated the rt workflow with some adjustments.